src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -1673,6 +1673,20 @@ public Double zscore(final String key, final String member) {
     return BuilderFactory.DOUBLE.build(client.getOne());
   }
 
+  @Override
+  public List<String> zpopmax(final String key) {
```

Comment:
Return type should be `Tuple`

Reply:
Now that you've mentioned it, of course it should be :) I'll make the changes shortly.

Reply:
> Return type should be `Tuple`

Could you please help with this inside `BinaryJedis` - I can't figure out what it should look like. There's no single Tuple being returned anywhere that I could find, just Set<Tuple>. 

```
  public List<byte[]> zpopmax(final byte[] key) {
    checkIsInMultiOrPipeline();
    client.zpopmax(key);
    return client.getBinaryMultiBulkReply();
  }
```

Reply:
There are 2 ways I can think of, right now.
1. Create `Set<Tuple>` from response (see other examples) and return the only element.
2. Create `List<byte[]>` of two elements and use `Tuple(byte[], Double)` constructor.

Reply:
How does this look under `BinaryJedis `?

```
  public Tuple zpopmax(final byte[] key) {
    checkIsInMultiOrPipeline();
    client.zpopmax(key);
    
    List<byte[]> result = client.getBinaryMultiBulkReply();
    
    return new Tuple(result.get(0), BuilderFactory.DOUBLE.build(result.get(1)));
  }
```

And what about the `PipelineBase` implementation?
  
```
  public Response<List<String>> zpopmax(final String key) {
    getClient(key).zpopmax(key);
    return getResponse(BuilderFactory.STRING_LIST);
  }
```

There already is a `BuilderFactory.Tuple_ZSET` type, but it returns a `Set<Tuple>`.

I see 2 options here: 
a) Make a new builder that returns a simple `Tuple`?
b) Use a Set<Tuple> for the zpopmax(String key) variant, that will always only have 1 element?

Reply:
Made the changes, please have a look.